common_sqlite/src/error.rs

```diff
@@ -22,12 +22,45 @@
 
 use std::num::TryFromIntError;
 
+use diesel::{self, r2d2};
+use tari_utilities::message_format::MessageFormatError;
 use thiserror::Error;
+use tokio::task;
 
 #[derive(Debug, Error)]
 pub enum SqliteStorageError {
-    #[error("Diesel R2d2 error")]
-    DieselR2d2Error(String),
     #[error("Poolsize is too big")]
     PoolSize(#[from] TryFromIntError),
+    #[error("Operation not supported")]
+    OperationNotSupported,
+    #[error("Conversion error: `{0}`")]
+    ConversionError(String),
+    #[error("Database error: `{0}`")]
+    R2d2Error(#[from] r2d2::Error),
+    #[error("Database error: `{0}`")]
+    DieselR2d2Error(String),
+    #[error("Database error: `{0}`")]
+    DieselConnectionError(#[from] diesel::ConnectionError),
+    #[error("Database error: `{0}`")]
+    DatabaseMigrationError(String),
+    #[error("Database error: `{0}`")]
+    BlockingTaskSpawnError(String),
+}
+
+#[derive(Debug, Error)]
+pub enum StorageError {
```

Comment:
At some point we can merge this and the above. The diesel errors are too general to have as "StorageError"